packages/components/src/systems/PatientMedHistory/index.tsx

```diff
@@ -62,17 +82,34 @@ export default function PatientMedHistory(props: PatientMedHistoryProps) {
 
   const baseURL = createMemo(() => `${client?.clinicalUrl}/prescriptions/`);
 
-  const fetchPatient = async () => {
-    const { data } = await client!.apollo.query({
-      query: GET_PATIENT_MED_HISTORY,
-      variables: { id: props.patientId }
-    });
-    if (data?.patient?.medicationHistory) {
-      setMedHistory(data.patient.medicationHistory);
-    } else {
+  const queryOptions = createMemo(() => ({
+    variables: { id: props.patientId },
+    client: client!.apollo
+  }));
```

Comment:
Why separate this out?

Reply:
because solid lifecycle wouldn't catch the patientId updating which would update the query, this will make that happen